var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -778,3 +836,106 @@ def check(self):
+
+        if "+cuda" in spec and len(spec.variants["cuda_arch"].value) > 1:
+            raise InstallError("CP2K supports only one cuda_arch at a time")
+
+        if "+rocm" in spec and len(spec.variants["amdgpu_target"].value) > 1:
+            raise InstallError("CP2K supports only one amdgpu_arch at a time")
+
+        args = []
+
+        if spec.satisfies("+cuda"):
+            cuda_arch = self.spec.variants["cuda_arch"].value[0]
+
+            gpu_map = {
+                "35": "K40",
+                "37": "K80",
+                "60": "P100",
+                "70": "V100",
+                "80": "A100",
+            }
+
+            gpuver = gpu_map[cuda_arch]
+
+            args += ["-DCP2K_WITH_GPU=%s" % gpuver, "-DCP2K_USE_ACCEL=cuda"]
+
+        if spec.satisfies("+rocm"):
+            amd_arch = self.spec.variants["amdgpu_target"].value[0]
+
+            gpuver = {"gfx906": "Mi50", "gfx908": "Mi100", "gfx90a": "Mi250x"}[amd_arch]
+
+            args += ["-DCP2K_WITH_GPU={0}".format(gpuver), "-DCP2K_USE_ACCEL=hip"]
+
+        args += [
+                self.define_from_variant("CP2K_USE_ELPA", "elpa"),
+                self.define_from_variant("CP2K_USE_LIBINT", "libint"),
+                self.define_from_variant("CP2K_USE_SIRIUS", "sirius"),
+                self.define_from_variant("CP2K_USE_SPLA", "spla"),
+                self.define_from_variant("CP2K_USE_COSMA", "cosma"),
+                self.define_from_variant("CP2K_USE_LIBXC", "libxc"),
+            ]
+
+        if spec.satisfies("+pytorch"):
+            args += ["-DCP2K_USE_LIBTORCH=ON"]
+
+        args += [
+                self.define_from_variant("CP2K_USE_METIS", "metis"),
+                self.define_from_variant("CP2K_USE_PLUMED", "plumed"),
+                self.define_from_variant("CP2K_USE_SPGLIB", "spglib"),
+                self.define_from_variant("CP2K_USE_VORI", "libvori"),
+                self.define_from_variant("CP2K_USE_SUPERLU", "superlu-dist"),
+                self.define_from_variant("CP2K_USE_SPLA", "spla"),
+                self.define_from_variant("CP2K_USE_QUIP", "quip"),
+            ]
+
+        # we force the use elpa openmp threading support. might need to be revisited though
+        if (spec.satisfies("+openmp") and spec.satisfies("+elpa")) or spec.satisfies("^elpa+openmp"):
+            args += [ "-DCP2K_ENABLE_ELPA_OPENMP_SUPPORT=ON" ]
+
+        if "spla" in spec and (spec.satisfies("+cuda") or spec.satisfies("+rocm")):
+            args += [ "-DCP2K_USE_SPLA_GEMM_OFFLOADING=ON" ]
+
+        if "fftw" in spec:
+            args += [ "-DCP2K_USE_FFTW=ON" ]
+
+        with when("smm=libxsmm"):
+            args += [ "-DCP2K_USE_LIBXSMM=ON" ]
+
+        if spec["blas"].name in ["intel-mkl", "intel-parallel-studio", "cray-libsci", "openblas"]:
+            if spec["blas"].name in ["intel-mkl", "intel-parallel-studio"]:
+                args += [ "-DCP2K_BLAS_VENDOR=MKL" ]
+                args += [ "-DCP2K_SCALAPACK_VENDOR=MKL" ]
```

Comment:
Please separate blas and scalapack here, on macOS for example there is no Intel MPI and no ScaLAPACK, hence the combination of MKL for BLAS/LAPACK and netlib-scalapack for scalapack.

Reply:
In general I wonder whether this kind of explicit spec for the scientific libs is still necessary with CMake or whether Spack provides a restricted enough environment for CMake to auto-discover them correctly?

Reply:
Yes cmake autodetects them automatically. It will take the first one it finds on a system I usually specify things to help cmake finding the dependencies but in the spack case this is known. 

In some case it is necessary to specify these (essl for instance). I have for instance five different implementations of blas / lapack on my system. So specifying one version allows me to test things. 

CP2K cmake build system will still generate a warning if it find mkl but scalapack is set to its default value. So When we know that we use MKL better specify the scalapack version as well. 

macos ? Not sure to understand what's wrong here.